main.go

```diff
@@ -44,6 +54,21 @@ func printVersion() {
 	fmt.Println("Metrics Exporter for pluggable transceivers on Linux based hosts / switches")
 }
 
+// compileRegexFlags compiles the cli regex flags into the global variables
+// and returns an error if the regex is invalid
+func compileRegexFlags() error {
```

Comment:
We might want to just not set excludeInterfacesRegexCompiled and includeInterfacesRegexCompiled, if there were no "real" input, e.g. an empty string. This cleans a check further down the line.

Reply:
I think we cannot do that.
If the regex will be compiled with an empty string, the `.MatchString()` function will match everything. So we need to check for the length of the string somewhere.

If we do that at startup we could not put a compiled regex in the the global variables (e.g. `excludeInterfacesRegexCompiled`). But then a nil pointer dereference at `.MatchString()` happens. The variable with the initiated `&regexp.Regexp{}` itself is not nil so we cannot check on that.

```bash
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x6d3e5e]

goroutine 23 [running]:
regexp.(*machine).match(0xc0000ce0f0, {0x98fe20, 0xc0000ce1b0}, 0x0)
        /nix/store/l9n531fcx4grk3wvfbigya3p66scj61h-go-1.20.6/share/go/src/regexp/exec.go:222 +0x39e
regexp.(*Regexp).doExecute(0xc8b6e0, {0x0?, 0x0}, {0x0, 0x0, 0x0}, {0xc0000c4000, 0x7}, 0x0?, 0x0, ...)
        /nix/store/l9n531fcx4grk3wvfbigya3p66scj61h-go-1.20.6/share/go/src/regexp/exec.go:542 +0x319
regexp.(*Regexp).doMatch(...)
        /nix/store/l9n531fcx4grk3wvfbigya3p66scj61h-go-1.20.6/share/go/src/regexp/exec.go:514
regexp.(*Regexp).MatchString(...)
        /nix/store/l9n531fcx4grk3wvfbigya3p66scj61h-go-1.20.6/share/go/src/regexp/regexp.go:533
github.com/wobcom/transceiver-exporter/transceiver-collector.(*TransceiverCollector).getMonitoredInterfaces(0xc0000b2230)
```

Reply:
Hi, 

sorry for missing your ping. 
I think, we roll with this for now.